neqo-http3/src/server.rs

```diff
@@ -1267,6 +1267,7 @@ mod tests {
         let out = peer_conn.process(None, now());
         hconn.process(out.as_dgram_ref(), now());
 
+        #[allow(clippy::mutable_key_type)]
```

Comment:
This is just a test, so this is probably OK, but this could be keyed on stream ID instead.  It would be better if it was.  I think that you can just use `stream.stream_id()` in those places that you use the stream.